src/test/java/com/easypost/ErrorTest.java

```diff
@@ -34,4 +55,93 @@ public void testError() throws EasyPostException {
+            put(401, UnauthorizedError.class);
+            put(402, PaymentError.class);
+            put(403, ForbiddenError.class);
+            put(404, NotFoundError.class);
+            put(405, MethodNotAllowedError.class);
+            put(408, TimeoutError.class);
+            put(422, InvalidRequestError.class);
+            put(429, RateLimitError.class);
+            put(444, UnknownApiError.class);
+            put(500, InternalServerError.class);
+            put(503, ServiceUnavailablError.class);
+            put(504, GatewayTimeoutError.class);
+        }};
+
+        for (Map.Entry<Integer, Class<?>> entry: apiErrorsMap.entrySet()) {
+            EasyPostException exception = assertThrows(EasyPostException.class,
+                () -> EasyPostResource.handleAPIError("{}", entry.getKey()));
+
+            assertEquals(Constants.API_DID_NOT_RETURN_ERROR_DETAILS, exception.getMessage());
+            assertEquals("NO RESPONSE CODE", exception.getCode());
+            assertEquals(entry.getKey(), exception.getStatusCode());
+            assertInstanceOf(entry.getValue(), exception);
+        }
+    }
+
+    /**
+     * Test parsing error message.
+     *
+     * @throws EasyPostException
+     */
+    @Test
+    public void testExceptionErrorMessageParsing() throws EasyPostException {
+        String errorMessageStringJson =
+            "{\"error\": {\"code\": \"ERROR_CODE\", \"message\": \"ERROR_MESSAGE_1\", \"errors\": []}}";
+        EasyPostException exception = assertThrows(EasyPostException.class,
+            () -> EasyPostResource.handleAPIError(errorMessageStringJson, 400));
+
+        assertEquals("ERROR_MESSAGE_1", exception.getMessage());
+    }
+
+    /**
+     * Test parsing error message that is an array.
+     *
+     * @throws EasyPostException
+     */
+    @Test
+    public void testExceptionErrorArrayParsing() throws EasyPostException {
+        String errorMessageArrayJson = "{\"error\": {\"code\": \"ERROR_CODE\", \"message\":" +
+            "[\"ERROR_MESSAGE_1\", \"ERROR_MESSAGE_2\"], \"errors\": []}}";
+
+        EasyPostException exception = assertThrows(EasyPostException.class,
+            () -> EasyPostResource.handleAPIError(errorMessageArrayJson, 400));
+
+        assertEquals("ERROR_MESSAGE_1, ERROR_MESSAGE_2", exception.getMessage());
+    }
+
+    /**
+     * Test creating invalid address creation to see if the error has correct properties.
+     *
+     * @throws EasyPostException
+     */
+    @Test
+    public void testInvalidAddressCreation() throws EasyPostException {
+        vcr.setUpTest("error_address_creation");
+        EasyPostException exception = assertThrows(EasyPostException.class,
+                () -> Address.createAndVerify(null));
+
+        assertEquals("PARAMETER.REQUIRED", exception.getCode());
+        assertEquals(422, exception.getStatusCode());
+        assertEquals("Missing required parameter.", exception.getMessage());
+    }
```

Comment:
Issue: Please move this to the `Address` test file like other libs. It's intended to test that even though we ripped out some logic some time ago about how errors were handled for addresses that it still gets thrown correctly.

Reply:
resolved